src/backend/distributed/executor/subplan_execution.c

```diff
@@ -56,7 +56,7 @@ ExecuteSubPlans(DistributedPlan *distributedPlan)
 	 */
 	BeginOrContinueCoordinatedTransaction();
 
-	nodeList = ActiveReadableNodeList();
+	nodeList = ActiveReadableWorkerNodeList();
```

Comment:
I think this is only true if we merge #3170, otherwise we could fail to support the following because the anchor shard might be the coordinator for the reference table:

```SQL
SET citus.task_assignment_policy TO "round-robin";
WITH cte1 AS (SELECT value_1, random() FROM users_table) SELECT count(*) FROM users_table_ref JOIN cte1 USING (value_1);
```

Anyway, just wanted to mention so that we don't forget to test on the main PR

Reply:
Makes sense. I'll merge it to master after merging this PR to master.